libnavigation-core/src/main/java/com/mapbox/navigation/core/stops/ArrivalProgressObserver.kt

```diff
@@ -11,24 +11,39 @@ internal class ArrivalProgressObserver(
 ) : RouteProgressObserver {
 
     private var arrivalController: ArrivalController = AutoArrivalController()
+    private val arrivalObservers: MutableSet<ArrivalObserver> = mutableSetOf()
     private var arrivedForRoute = false
 
     fun attach(arrivalController: ArrivalController) {
         this.arrivalController = arrivalController
     }
 
+    fun registerObserver(arrivalObserver: ArrivalObserver) {
+        arrivalObservers.add(arrivalObserver)
+    }
+
+    fun unregisterObserver(arrivalObserver: ArrivalObserver) {
+        arrivalObservers.remove(arrivalObserver)
+    }
+
     fun navigateNextRouteLeg(): Boolean {
-        val numberOfLegs = tripSession.getRouteProgress()?.route()?.legs()?.size
+        val routeProgress = tripSession.getRouteProgress()
+        val numberOfLegs = routeProgress?.route()?.legs()?.size
             ?: return false
-        val legIndex = tripSession.getRouteProgress()?.currentLegProgress()?.legIndex()
+        val legProgress = routeProgress.currentLegProgress()
+        val legIndex = legProgress?.legIndex()
             ?: return false
         val nextLegIndex = legIndex + 1
-        return if (nextLegIndex < numberOfLegs) {
+        val nextLegStarted = if (nextLegIndex < numberOfLegs) {
             val navigationStatus = tripSession.updateLegIndex(nextLegIndex)
             return nextLegIndex == navigationStatus.legIndex
         } else {
-            true
+            false
```

Comment:
guess this was a bug.. adding a test

Reply:
added https://github.com/mapbox/mapbox-navigation-android/pull/2855/files#diff-015f94bcfe8f96fa0f98b944014cdbbaR355